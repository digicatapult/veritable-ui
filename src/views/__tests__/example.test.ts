import { describe, it } from 'mocha'

import ExampleTemplates from '../example'

describe('RootTemplates', async () => {
  let expect: Chai.ExpectStatic
  before(async () => {
    expect = (await import('chai')).expect
  })

  describe('Root', () => {
    it('should render root page', async () => {
      const templates = new ExampleTemplates()
      const rendered = await templates.Root('title', 0)
      expect(rendered).to.matchSnapshot()
    })

    it('should escape html in title', async () => {
      const templates = new ExampleTemplates()
      const rendered = await templates.Root('<div>Malicious Content</div>', 0)
      expect(rendered).to.matchSnapshot()
    })

    it('should render counter 1 on second load', async () => {
      const templates = new ExampleTemplates()
      const rendered = await templates.Root('title', 1)
      expect(rendered).to.matchSnapshot()
    })
  })

  describe('Counter', () => {
    it('should render the counter with value 0', async () => {
      const templates = new ExampleTemplates()
      const rendered = await templates.Counter({ count: 0 })
      expect(rendered).to.matchSnapshot()
    })

    it('should render the counter on second call with value 1', async () => {
      const templates = new ExampleTemplates()
      const rendered = await templates.Counter({ count: 1 })
      expect(rendered).to.matchSnapshot()
    })
  })

  describe('Button', () => {
    it('should render the enabled button', async () => {
      const templates = new ExampleTemplates()
      const rendered = await templates.Button({ disabled: false })
      expect(rendered).to.matchSnapshot()
    })

    it('should render the disabled button', async () => {
      const templates = new ExampleTemplates()
      const rendered = await templates.Button({ disabled: true })
      expect(rendered).to.matchSnapshot()
    })
  })
})
