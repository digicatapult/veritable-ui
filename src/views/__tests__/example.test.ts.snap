// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootTemplates Button should render the disabled button 1`] = `"<div class=\\"button-group\\"><button disabled hx-target=\\"closest .button-group\\" hx-trigger=\\"counter-loaded from:body\\" hx-get=\\"/example/button\\" hx-swap=\\"outerHTML\\">Click me!</button></div>"`;

exports[`RootTemplates Button should render the enabled button 1`] = `"<div class=\\"button-group\\"><button hx-target=\\"closest .button-group\\" hx-post=\\"/example/button\\" hx-swap=\\"outerHTML\\">Click me!</button></div>"`;

exports[`RootTemplates Counter should render the counter on second call with value 1 1`] = `"<div id=\\"counter\\" hx-get=\\"/example/counter\\" hx-trigger=\\"button-click from:body\\" hx-swap=\\"outerHTML\\"><span>1</span><img class=\\"spinner htmx-indicator\\" src=\\"/public/images/spinner.svg\\"/></div>"`;

exports[`RootTemplates Counter should render the counter with value 0 1`] = `"<div id=\\"counter\\" hx-get=\\"/example/counter\\" hx-trigger=\\"button-click from:body\\" hx-swap=\\"outerHTML\\"><span>0</span><img class=\\"spinner htmx-indicator\\" src=\\"/public/images/spinner.svg\\"/></div>"`;

exports[`RootTemplates Root should escape html in title 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"lib/htmx.org/htmx.min.js\\"></script><script src=\\"lib/htmx.org/ext/json-enc.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>&lt;div>Malicious Content&lt;/div></title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><div class=\\"flex-page side-bar\\"></div><div class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\"></h1><div class=\\"header nav\\"><div>home icon</div><span class=\\"url-separator\\">/</span><div></div></div></div><div class=\\"content main\\"><div id=\\"counter\\" hx-get=\\"/example/counter\\" hx-trigger=\\"button-click from:body\\" hx-swap=\\"outerHTML\\"><span>0</span><img class=\\"spinner htmx-indicator\\" src=\\"/public/images/spinner.svg\\"/></div><div class=\\"button-group\\"><button hx-target=\\"closest .button-group\\" hx-post=\\"/example/button\\" hx-swap=\\"outerHTML\\">Click me!</button></div></div></div></body></html>"`;

exports[`RootTemplates Root should render counter 1 on second load 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"lib/htmx.org/htmx.min.js\\"></script><script src=\\"lib/htmx.org/ext/json-enc.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>title</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><div class=\\"flex-page side-bar\\"></div><div class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\"></h1><div class=\\"header nav\\"><div>home icon</div><span class=\\"url-separator\\">/</span><div></div></div></div><div class=\\"content main\\"><div id=\\"counter\\" hx-get=\\"/example/counter\\" hx-trigger=\\"button-click from:body\\" hx-swap=\\"outerHTML\\"><span>1</span><img class=\\"spinner htmx-indicator\\" src=\\"/public/images/spinner.svg\\"/></div><div class=\\"button-group\\"><button hx-target=\\"closest .button-group\\" hx-post=\\"/example/button\\" hx-swap=\\"outerHTML\\">Click me!</button></div></div></div></body></html>"`;

exports[`RootTemplates Root should render root page 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"lib/htmx.org/htmx.min.js\\"></script><script src=\\"lib/htmx.org/ext/json-enc.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>title</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><div class=\\"flex-page side-bar\\"></div><div class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\"></h1><div class=\\"header nav\\"><div>home icon</div><span class=\\"url-separator\\">/</span><div></div></div></div><div class=\\"content main\\"><div id=\\"counter\\" hx-get=\\"/example/counter\\" hx-trigger=\\"button-click from:body\\" hx-swap=\\"outerHTML\\"><span>0</span><img class=\\"spinner htmx-indicator\\" src=\\"/public/images/spinner.svg\\"/></div><div class=\\"button-group\\"><button hx-target=\\"closest .button-group\\" hx-post=\\"/example/button\\" hx-swap=\\"outerHTML\\">Click me!</button></div></div></div></body></html>"`;
